flink-libraries/flink-table/src/main/scala/org/apache/flink/table/validate/FunctionCatalog.scala

```diff
@@ -107,6 +107,18 @@ class FunctionCatalog {
         val function = aggregateFunction.getFunction
         AggFunctionCall(function, children)
 
+      // concat_ws
+      case concat_ws if classOf[ConcatWs].isAssignableFrom(concat_ws) =>
```

Comment:
I'm not sure whether this is a good way. I prefer to add an additional try (for constructor `classOf[Expression], classOf[Seq[_]]`) in general expression construction.